### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by mrexodia**
I did some improvements in the GUI and also resolved issue #2

### CODE REVIEW

File: src/nc/gui/MainWindow.cpp [hunk 1]
```diff
@@ -88,7 +88,8 @@ MainWindow::MainWindow(Branding branding, QWidget *parent):
 
     logToken_ = LogToken(logger);
 
-    settings_ = new QSettings(branding_.organizationName(), branding_.applicationName(), this);
+    settings_ = new QSettings(QString(branding.applicationName() + ".ini"), QSettings::IniFormat, this);
```
Comment (yegord) [inline src/nc/gui/MainWindow.cpp]: I guess, this will put settings into the current directory, which generally is not going to be writable by the user.

Comment (mrexodia) [inline src/nc/gui/MainWindow.cpp]: Yea, that's correct. But why wouldn't the current directory be readable? Anyways, I can revert the commit without problems (doesn't matter to me). Do you agree with the rest of the changes?

Comment (yegord) [inline src/nc/gui/MainWindow.cpp]: Well, on Windows programs are usually installed to Program Files, which is not writable by usual users. And these programs are typically launched with Program Files/Program Name being their current directory. Nonetheless, usual users should be able to store settings.

If you want to allow the programmer to store options somewhere else, you can pass QSettings * to the MainWindow constructor. If nullptr is passed, the settings must be constructed the way they are now constructed. I would accept such a change.

Concerning the rest, I have not read everything yet. :)

**Comment by mrexodia** [comment]
I removed the settings commit

**Comment by mrexodia** [comment]
Right click and do search next. It will crash.

**Comment by yegord** [comment]
It actually will crash even after your patch, if I just push Ctrl-F immediately after the start.
I will push a fix shortly.

**Comment by mrexodia** [comment]
Okay, will you also fix the other things? The huge margin in the goto line
and search widget and also the test intrinsic.

**Comment by yegord** [comment]
Your commit for sf in test was pushed yesterday.
Concerning margin — F just on the border look ugly: http://derevenets.com/tmp/Find.png
Maybe we can set zero margin on the right but not on the left...

**Comment by yegord** [comment]
Actually, the same situation with buttons: they touch the window border if you set zero margin.
I think, I already wanted to set margin to zero some time ago, and this was the reason I did not.

**Comment by mrexodia** [comment]
Just set a margin to the actual widget. I'll come up with something nice
later on.

**Comment by mrexodia** [comment]
i'll do a new pullreq for the changes with the margin

**Pull request closed without merging**